tests/workbenches/utils.py

```diff
@@ -0,0 +1,221 @@
+#
+# Copyright Skodjob authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+#
+from __future__ import annotations
+
+import logging
+import time
+import traceback
+from contextlib import contextmanager
+from typing import Any
+
+from typing import Callable, Generator
+
+import kubernetes.dynamic
+from kubernetes.dynamic import DynamicClient, ResourceField
+
+import ocp_resources.pod
+import ocp_resources.resource
+
+
+class TestFrameConstants:
+    GLOBAL_POLL_INTERVAL_MEDIUM = 10
+
+
+class OcpResourceManager:
+    def __init__(self) -> None:
+        self.resources: list[ocp_resources.resource.Resource] = []
+
+    def enter(self, resource: ocp_resources.resource.Resource) -> None:
+        resource.__enter__()
+        self.resources.append(resource)
+
+    def __enter__(self) -> OcpResourceManager:
+        return self
+
+    def __exit__(self, *args: tuple[Any, ...], **kwargs: dict[str, Any]) -> None:
+        exceptions = []
+        for resource in reversed(self.resources):
+            try:
+                resource.__exit__(*args, **kwargs)
+            except Exception as e:
+                exceptions.append(e)
+        if exceptions:
+            raise RuntimeError(f"Exceptions during cleanup: {exceptions}")
+
+
+class PodUtils:
```

Comment:
Same as above, if you use the Pod class you can already handle this without having to reimplement these checks

Reply:
done